src/distributed-transaction/percolator.md

```diff
@@ -0,0 +1,153 @@
+# Percolator
+
+TiKV supports the distributed transaction, which is inspired by Google's paper about *Percolator*. We will introduce it and how we make use of it in TiKV in this section.
+
+## What is Percolator?
+
+*Percolator* is a system used for incremental processing on very large data set, built by Google. Support for distributed transactions is only a part of it. We will introduce Percolator briefly here. You can view the full paper [here](https://ai.google/research/pubs/pub36726#), and if you are very familiar to this paper, you can skip this section and go to read [Percolator in TiKV](#Percolator-in-TiKV)
+
+Percolator is built based on Google's BigTable. BigTable is a distributed storage system that supports single-row transactions, based on which Percolator implements distributed transactions in ACID snapshot-isolation semantics. A column `c` of Percolator is actually divided into following internal columns of BigTable:
```

Comment:
```suggestion
Percolator is built based on Google's BigTable, a distributed storage system that supports single-row transactions. With similar logic, Percolator implements distributed transactions in ACID snapshot-isolation semantics. A column `c` of Percolator is actually divided into the following internal columns of BigTable:
```

Reply:
Not "with similar logic". BigTable doesn't support ACID distributed transaction, and Percolator implemented it based on BigTable. How do you think if I change this paragraph to this?
> Percolator is built based on Google's BigTable, a distributed storage system that supports single-row transactions. Percolator implements distributed transactions in ACID snapshot-isolation semantics, which is not supported by BigTable. ...

Reply:
Yeah it's accurate. At first I just wanted to avoid using "based on" twice in two sentences in a row. Go ahead and change it.